reason-react-native/src/apis/CameraRoll.md

```diff
@@ -130,7 +131,96 @@ type action =
   | SetPath(string)
   | SetPhotos(array(CameraRoll.photoIdentifier));
 
-let textStyle = Style.style(~textAlign=`center, ());
+let styles = Style.(
+  StyleSheet.create({
+    "container":
+      style(
+        ~flex=1.,
+        ~flexDirection=`column,
+        ~alignItems=`center,
+        ~justifyContent=`spaceBetween,
+        (),
+      ),
+    "getPhotosExample":
+      style(
+        ~height=pt(0.25 *. windowHeight),
+        ~justifyContent=`spaceBetween,
+        (),
+      ),
+    "saveToCameraRollExample":
+      style(
+        ~width=pt(0.8 *. windowWidth),
+        ~height=pt(0.6 *. windowHeight),
+        ~margin=pt(0.1 *. windowWidth),
+        ~justifyContent=`spaceAround,
+        (),
+      ),
+    "image":
+      style(
+        ~width=pt(0.8 *. windowWidth),
+        ~height=pt(0.56 *. windowWidth),
+        (),
+      ),
+    "imageContainer":
+      style(
+        ~width=pt(0.8 *. windowWidth),
+        ~height=pt(0.56 *. windowWidth),
+        ~borderWidth=1.,
+        ~justifyContent=`center,
+        (),
+      ),
+    "text": style(~textAlign=`center, ()),
+    "textInput":
+      style(
+        ~textAlign=`center,
+        ~borderRadius=8.,
+        ~padding=pt(4.),
+        ~borderWidth=1.,
+        ~backgroundColor=Color.linen,
+        (),
+      ),
+    "thumbnail":
+      style(
+        ~width=pt(0.21 *. windowHeight),
+        ~height=pt(0.15 *. windowHeight),
+        (),
+      ),
+  })
+);
+
+let styledText = s => {
+  <Text style=styles##text> s->React.string </Text>;
+};
+
+let thumbnails = photos => {
+  <View style={Style.style(~height=Style.pt(0.15 *. windowHeight), ())}>
+    <ScrollView horizontal=true>
+      {React.array(
+         Array.mapi(
```

Comment:
Small thing, but in the spirit of zero I'd rather use `Js.Array.mapi`.

Reply:
Thanks!

Reply:
I would ever go mo aggresive using `Belt.Array.mapWithIndex` with fastpipe. It seems that new comers (at least the one I know) find this more natural

```reason
photos
->Belt.Array.mapWithIndex((i, s) => <Image /* .... */ />)
->React.array
```

Reply:
That's fine, too, and it's what we are doing in our projects (we are using Belt exclusively). If you are not using Belt, it brings an additional dependency into your project though.

BTW, there is now a pipe-first-compatible Js.Array2, too.

Reply:
I wasn't too sure if using pipes would be confusing for newcomers (OK, it's already used here with `React.string`, but that's very simple), but it's easier to read for sure.

I think it's better to stick with Js.Array which works with `|>`) or Js.Array2 (I wasn't aware of this) if `->` is more familiar in general. Either simply maps JS native `map` method of the array. 

I like to use `Belt` when it's useful (no out of bound errors when getting items from arrays, for example), but I don't see what `mapWithIndex` has over `mapi`.

So, `->` or `|>`?

Reply:
I am leaning towards `->`, but it's not a big deal, I'll approve the PR as it is now.

Reply:
I probably would not have noticed `Js.Array2` for at least a bit longer had it not been for your comment. So I'll use `Js.Array2` to draw attention to that module.

Reply:
not sure reasonml/buckelscript teams want to promote *2 modules...
I would definitely go with `->` & Belt.
Learning Belt is a better idea, having native compilation in mind :)

Reply:
Yeah, the *2 modules are hopefully only a temporary solution until everything is pipe first.

Ok, so let's go with Belt + pipe first!